js/ui/select_box.js

```diff
@@ -109,7 +110,11 @@ var SelectBox = DropDownList.inherit({
                 this._cancelEditing();
             },
             enter: function(e) {
-                if(this._input().val() === "" && this.option("value") && this.option("allowClearing")) {
+                var inputText = this._input().val();
+                var inputTextIsEmpty = isEmpty(inputText);
+                var inputHasCustomValue = this._list && !this._list.option("focusedElement");
```

Comment:
What if `focusStateEnabled` is false? All mobile devices have no focusedElement in the list. Please check this